examples/gno.land/p/demo/uassert/uassert.gno

```diff
@@ -416,3 +419,514 @@ func NotEmpty(t TestingT, obj interface{}, msgs ...string) bool {
+	}
+	return true
+}
+
+// IsType asserts that two objects are of the same type.
+func IsType(t TestingT, expectedType, actual interface{}, msgs ...string) bool {
+	t.Helper()
+	switch expectedType.(type) {
+	case bool:
+		if _, ok := actual.(bool); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type bool but got %T", actual)
+		}
+	case int:
+		if _, ok := actual.(int); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type int but got %T", actual)
+		}
+	case int8:
+		if _, ok := actual.(int8); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type int8 but got %T", actual)
+		}
+	case int16:
+		if _, ok := actual.(int16); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type int16 but got %T", actual)
+		}
+	case int32:
+		if _, ok := actual.(int32); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type int32 but got %T", actual)
+		}
+	case int64:
+		if _, ok := actual.(int64); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type int64 but got %T", actual)
+		}
+	case uint:
+		if _, ok := actual.(uint); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type uint but got %T", actual)
+		}
+	case uint8:
+		if _, ok := actual.(uint8); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type uint8 but got %T", actual)
+		}
+	case uint16:
+		if _, ok := actual.(uint16); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type uint16 but got %T", actual)
+		}
+	case uint32:
+		if _, ok := actual.(uint32); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type uint32 but got %T", actual)
+		}
+	case uint64:
+		if _, ok := actual.(uint64); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type uint64 but got %T", actual)
+		}
+	case string:
+		if _, ok := actual.(string); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type string but got %T", actual)
+		}
+	case std.Address:
+		if _, ok := actual.(std.Address); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type std.Address but got %T", actual)
+		}
+	default:
+		return fail(t, msgs, "uassert.IsType: unsupported type %T", expectedType)
+	}
+	return true
+}
+
+// Regexp asserts that a specified regexp matches matches a string.
+func Regexp(t TestingT, rx interface{}, str interface{}, msgs ...string) bool {
+	t.Helper()
+	var r *regexp.Regexp
+	switch v := rx.(type) {
+	case *regexp.Regexp:
+		r = v
+	case string:
+		r = regexp.MustCompile(v)
+	default:
+		return fail(t, msgs, "uassert.Regexp: invalid regexp type")
+	}
+
+	s, ok := str.(string)
+	if !ok {
+		return fail(t, msgs, "uassert.Regexp: invalid string type")
+	}
+
+	if !r.MatchString(s) {
+		return fail(t, msgs, "uassert.Regexp: \"%s\" does not match \"%s\"", s, r.String())
+	}
+	return true
+}
+
+// ElementsMatch asserts that two lists contain the same elements, regardless of order.
+func ElementsMatch(t TestingT, listA, listB []interface{}, msgs ...string) bool {
```

Comment:
This signature differs from how it works in testify and how most users might expect it to work.  I believe that the natural expectation is to pass two slice to this function and have it check if the elements match. The current implementation requires the slices to be transformed into slices of interfaces before this can be used; if that's the case, they might as well just check equality while iterating over the slices instead of building the interface slices.

So if we want a function like `Elements` match, it should accept `interface{}` types and then:
1. Check to make sure that the slice types match
2. Check to make sure they have the same length
3. Check to make sure each of their elements are equal

Even then, the implementation is incomplete because it assumes primitive type slices; there would need to be deeper equality checks done to do this correctly.